src/golang/lib/job/config.go

```diff
@@ -84,3 +97,173 @@ func RegisterGobTypes() {
 func init() {
 	RegisterGobTypes()
 }
+
+func GenerateNewJobManager(
+	ctx context.Context,
+	engineConfig shared.EngineConfig,
+	storageConfig *shared.StorageConfig,
+	aqPath string,
+	vault vault.Vault,
+) (JobManager, error) {
+	jobConfig, err := GenerateJobManagerConfig(
+		ctx,
+		engineConfig,
+		storageConfig,
+		aqPath,
+		vault,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to generate JobManagerConfig.")
+	}
+
+	jobManager, err := NewJobManager(jobConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to create JobManager.")
+	}
+	return jobManager, nil
+}
+
+func GenerateJobManagerConfig(
+	ctx context.Context,
+	engineConfig shared.EngineConfig,
+	storageConfig *shared.StorageConfig,
+	aqPath string,
+	vault vault.Vault,
+) (Config, error) {
+	switch engineConfig.Type {
+	case shared.AqueductEngineType:
+		return &ProcessConfig{
+			BinaryDir:          path.Join(aqPath, BinaryDir),
+			OperatorStorageDir: path.Join(aqPath, OperatorStorageDir),
+		}, nil
+	case shared.K8sEngineType:
+		if storageConfig.Type != shared.S3StorageType && storageConfig.Type != shared.GCSStorageType {
+			return nil, errors.New("Must use S3 or GCS storage config for K8s engine.")
+		}
+
+		var awsAccessKeyId, awsSecretAccessKey string
+		if storageConfig.Type == shared.S3StorageType {
+			keyId, secretKey, err := extractAwsCredentials(storageConfig.S3Config)
+			if err != nil {
+				return nil, errors.Wrap(err, "Unable to extract AWS credentials from file.")
+			}
+
+			awsAccessKeyId = keyId
+			awsSecretAccessKey = secretKey
+		}
+
+		k8sIntegrationId := engineConfig.K8sConfig.IntegrationId
+		config, err := auth.ReadConfigFromSecret(ctx, k8sIntegrationId, vault)
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to read config from vault.")
+		}
+		k8sConfig, err := lib_utils.ParseK8sConfig(config)
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to get integration.")
+		}
+		return &K8sJobManagerConfig{
+			KubeconfigPath:     k8sConfig.KubeconfigPath,
+			ClusterName:        k8sConfig.ClusterName,
+			UseSameCluster:     bool(k8sConfig.UseSameCluster),
+			AwsAccessKeyId:     awsAccessKeyId,
+			AwsSecretAccessKey: awsSecretAccessKey,
+			AwsRegion:          DefaultAwsRegion,
+		}, nil
+	case shared.LambdaEngineType:
```

Comment:
Not restricted to this PR, I'm curious if we can remove / reduce the need for such switches if we do a better job refactoring the interfaces?